### PR TITLE
Update to tantivy v0.21.0

### PR DESCRIPTION
**Pull request by GodTamIt**
This change bumps the version of `tantivy` to v0.21.0.

Notably in this change is the addition of `Order` to specify order_by sort order for fast fields that was introduced in the new version.

### CODE REVIEW

File: src/searcher.rs
Comment (cjrh) [inline src/searcher.rs:160]: How strongly do you feel that this should be called `order_by_order` instead of just `order`?

File: src/searcher.rs
Comment (cjrh) [inline src/searcher.rs:151]: Just checking my understanding: this must be `Desc` because the function it replaces, `order_by_u64_field` was also descending, according to these docs: https://docs.rs/tantivy/0.19.2/tantivy/collector/struct.TopDocs.html#method.order_by_u64_field

Is that correct?

File: src/searcher.rs
Comment (GodTamIt) [reply src/searcher.rs:160]: Not particularly strongly here. Just thought it'd be more specific about which other parameter its associated with but I don't anticipate another ordering field any time soon?

File: tests/tantivy_test.py
Comment (cjrh) [inline tests/tantivy_test.py:574]: Hmm, why has this changed back?

File: src/searcher.rs
Comment (GodTamIt) [reply src/searcher.rs:151]: Yep, that's correct. It allows for code compatibility with the new version, which I defaulted to since there didn't seem to otherwise be a good reason to change the default.

File: tests/tantivy_test.py
Comment (cjrh) [reply tests/tantivy_test.py:574]: The writer should have been created with a large enough memory limit to allow 1 segment:

```python
writer = index.writer(10_000_000, 1)
```

^^ This is the writer inside the `create_index()` that is being called from `ram_index`.  I wonder if this change is related to the tantivy upgrade itself?

**Comment by cjrh** [comment]
Loving your work Chris 👍🏼

File: src/searcher.rs
Comment (cjrh) [reply src/searcher.rs:160]: I can live with `order_by_order`, but it feels "unpythonlike" 😄 .  We can hear from the other maintainers how they feel.

File: tests/tantivy_test.py
Comment (cjrh) [reply tests/tantivy_test.py:574]: @GodTamIt Could you test whether increasing that 10M brings the `num_segments` back down to 1?

Comment (GodTamIt) [reply tests/tantivy_test.py]: Looks like a small behavior change. Increasing the memory limit seems to have fixed the issue.

Comment (adamreichold) [reply tests/tantivy_test.py]: The memory accounting in Tantivy was improved and now tracks memory that was not included before, hence the same workload apparently needs more memory. (Actually, it needs the same, but the limit now includes more of the actual working set.)

File: src/searcher.rs
Comment (adamreichold) [reply src/searcher.rs:160]: To add yet another name to consider, how about `order_in_direction` (and matching `enum Direction`)?

File: src/searcher.rs
Comment (adamreichold) [inline src/searcher.rs:46]: Could we add doc comments to spell these out as `Ascending` and `Descending` as non-native speakers might have trouble recognizing the abbreviations.

Comment (GodTamIt) [reply src/searcher.rs:48]: I added doc comments above each discriminant. Let me know if that works

File: src/searcher.rs
Comment (GodTamIt) [reply src/searcher.rs]: I went with `order` (and a matching `enum Order`). I have a strong preference for an API surface and naming convention that closely mimics the original `tantivy` APIs, since I often go to those docs to see how things work in these bindings (as an outside user) :)

Comment (cjrh) [reply src/searcher.rs]: Same. I wish that `order_by_field` was just called `order_by`, but we can't easily change that now probably.

**Review by cjrh — APPROVED** [review]
I'm happy but I'd like to wait for @adamreichold to approve also.

**Review by adamreichold — APPROVED** [review]

**Pull request merged**
